### PR TITLE
ZENKO-833: add ingestion param for BucketInfo

### PR DESCRIPTION
**Pull request by JianqinWang**
The ingestion flag will be sent in the locationConstraint string as `<location name>:ingest`.
The standard location constraint string looks like `us-east-1`. With the ingestion flag, it would look like `us-east-1:ingest`.

### CODE REVIEW

File: lib/models/BucketInfo.js [hunk 1]
```diff
@@ -97,6 +100,11 @@ class BucketInfo {
         }
         if (locationConstraint) {
             assert.strictEqual(typeof locationConstraint, 'string');
+            ingestion = locationConstraint.split(zenkoSeparator)[1];
+        }
+        if (ingestionConfig) {
+            ingestion = ingestionConfig;
+            assert.strictEqual(typeof ingestion, 'string');
```
Comment (philipyoo) [inline lib/models/BucketInfo.js]: when will `locationConstraint` not have the ingestion flag but will have `ingestionConfig`?

File: lib/models/BucketInfo.js [hunk 2]
```diff
@@ -147,14 +155,16 @@ class BucketInfo {
...
+        this._locationConstraint = locationConstraint ?
+            locationConstraint.split(zenkoSeparator)[0] : null;
         this._readLocationConstraint = readLocationConstraint || null;
         this._websiteConfiguration = websiteConfiguration || null;
         this._replicationConfiguration = replicationConfiguration || null;
         this._cors = cors || null;
         this._lifecycleConfiguration = lifecycleConfiguration || null;
         this._uid = uid || uuid();
         this._isNFS = isNFS || null;
+        this._ingestion = ingestion ? { status: 'Enabled' } : null;
```
Comment (philipyoo) [inline lib/models/BucketInfo.js]: I think this is just fine, and can't think of any case where on bucket creation, ingestion is not enabled at the start, but maybe just to be safe, use the `setIngestion` method here to set this instance variable.

File: lib/models/BucketInfo.js (hunk 1 above)
Comment (JianqinWang) [reply lib/models/BucketInfo.js]: the ingestion flag would be its own separate value on a bucketInfo model, `locationConstraint` will be the carrier of the ingestion flag but will not store the ingestion value

File: lib/storage/metadata/mongoclient/MongoClientInterface.js [hunk 3]
```diff
@@ -1054,6 +1054,7 @@ class MongoClientInterface {
                             isVersioned:
                                 !!bucketInfo.getVersioningConfiguration(),
                             ownerCanonicalId: bucketInfo.getOwner(),
+                            ingestion: bucketInfo.isIngestionEnabled(),
```
Comment (rachedbenmustapha) [inline lib/storage/metadata/mongoclient/MongoClientInterface.js]: For the management layer's purposes, this should not change when ingestion is paused, what we need here is the nature of the bucket (ingest vs normal), not the current state.

**Comment by ghost** [comment]
There were the following issues with this Pull Request

* Commit: e24a5a14dbdb1ca50f134cebb822ae2c0b51e913
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

File: lib/storage/metadata/mongoclient/MongoClientInterface.js (hunk 3 above)
Comment (JianqinWang) [reply lib/storage/metadata/mongoclient/MongoClientInterface.js]: I've added a new function that would check if the bucket is/is not an ingestion bucket, and replaced it here :)

**Comment by ghost** [comment]
There were the following issues with this Pull Request

* Commit: da100967e46eb6ddbb0fdb36615403ac008497cd
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

File: lib/models/BucketInfo.js [hunk 4]
```diff
@@ -147,14 +155,16 @@ class BucketInfo {
...
+        this._locationConstraint = locationConstraint ?
+            locationConstraint.split(zenkoSeparator)[0] : null;
         this._readLocationConstraint = readLocationConstraint || null;
         this._websiteConfiguration = websiteConfiguration || null;
         this._replicationConfiguration = replicationConfiguration || null;
         this._cors = cors || null;
         this._lifecycleConfiguration = lifecycleConfiguration || null;
         this._uid = uid || uuid();
         this._isNFS = isNFS || null;
+        this._ingestion = ingestion ? { Status: 'Enabled' } : null;
```
Comment (rahulreddy) [inline lib/models/BucketInfo.js]: Let's keep this lowercased `{ status: enabled }` should be sufficient.

**Review by rahulreddy — CHANGES_REQUESTED** [review]
I would recommend setting 'status: { enabled/disabled }` lowercased. It's easier to manage when the code is either all uppercase or lowercase

File: lib/models/BucketInfo.js (hunk 4 above)
Comment (JianqinWang) [reply lib/models/BucketInfo.js]: I notice that for other configurations in BucketInfo, they have capitalization like this; would it be better to keep this uniform, or is it still better to go with lowercase?

File: lib/models/BucketInfo.js [hunk 5]
```diff
@@ -97,6 +100,11 @@ class BucketInfo {
         }
         if (locationConstraint) {
             assert.strictEqual(typeof locationConstraint, 'string');
+            ingestion = locationConstraint.split(zenkoSeparator)[1];
```
Comment (rahulreddy) [inline lib/models/BucketInfo.js]: Re-reviewing this...I don't think we need this here. Location constriant needs to be parsed in the API and the ingestionConfig you send as part of the constructor should be enough to set the bucket for ingestion enabled/disabled.

**Review by rahulreddy — CHANGES_REQUESTED** [review]

**Review by philipyoo — APPROVED** [review]

**Review by rahulreddy — APPROVED** [review]

**Pull request merged**